crm_timesheet/tests/test_account_analytic_line.py

```diff
@@ -1,32 +1,94 @@
 # Copyright 2017 Jairo Llopis <jairo.llopis@tecnativa.com>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo.tests.common import SavepointCase
+from datetime import date, datetime, timedelta
+from odoo.tests.common import TransactionCase
 
 
-class AccountAnalyticLineCase(SavepointCase):
-    @classmethod
-    def setUpClass(cls):
-        super(AccountAnalyticLineCase, cls).setUpClass()
-        Account = cls.env["account.analytic.account"]
-        Project = cls.env["project.project"]
-        cls.Line = cls.env["account.analytic.line"]
-        cls.account1 = Account.create({
+class AccountAnalyticLineCase(TransactionCase):
```

Comment:
Don't change test type

Reply:
I needed it to use the `self.uid` feature, which is not supported in `SavepointCase`.

It's needed because all operations must be made with a user that has an employee attached to it, and `__system__` doesn't have it.

Reply:
You can use `cls.env.user.id`. Any way, as always, I prefer to create all the needed data, so that's to say the user, and the employee (it's not too much).